src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -226,6 +275,117 @@ void handle_task_waiting(global_scheduler_state *state,
   free_object_size_hashmap(object_size_table);
 }
 
+/* Main pending task handler */
+void handle_task_waiting(global_scheduler_state *state,
+                         global_scheduler_policy_state *policy_state,
+                         task *task) {
+  task_spec *task_spec = task_task_spec(task);
+
+  CHECKM(task_spec != NULL,
+         "task wait handler encounted a task with NULL spec");
+#if (RAY_COMMON_LOG_LEVEL <= RAY_COMMON_DEBUG)
+  char buf[256];
```

Comment:
Using fixed size buffers seems a bit dangerous. We could use UT_string to get around this.